policy_packs/aws/ssm/enforce_install_packages_on_ec2_instances_using_ssm_package_installer/README.md

```diff
@@ -0,0 +1,205 @@
+
+This [policy pack](https://turbot.com/guardrails/docs/concepts/resources/smart-folders) is designed to allow package installation on instances with or without direct access to original package source and can help you configure the following settings for SSM documents:
+
+- Set the region ARN where the SSM resources will be deployed
+- Set the IAM role ARN to be used by SSM to run documents
+- Create three SSM documents in the specified region
+  - Turbot-LinuxPackageInstaller-PublicNetwork
+  - Turbot-LinuxPackageInstaller-PrivateNetwork
+  - Turbot-LinuxS3PackageInstaller
+
+## Documentation
+
+- **[Review Policy settings →](https://hub-guardrails-turbot-com-git-development-turbot.vercel.app/policy-packs/enforce_install_packages_on_ec2_instances_using_ssm_package_installer/settings)**
+- Using the SSM documents
+  - There are two entry point types:
+    - Turbot-LinuxPackageInstaller-PublicNetwork
+    - Turbot-LinuxPackageInstaller-PrivateNetwork
+
+### Turbot-LinuxPackageInstaller-PublicNetwork
+
+Use this entry point type when there is direct access to the package source.
+Using this method, all that is required to the URL of the package to install.
+
+### Turbot-LinuxPackageInstaller-PrivateNetwork
+
+Use this entry point type when there is no direct access to the package source.
+Using this method, it is required that the URL of the package to install and a valid S3 bucket which be used to cache the package.
+
+### Turbot-LinuxS3PackageInstaller
+
+This document should never be directly run but is a helper document to install the package.
+
+### Current limitations
+
+- Currently only Ubuntu is supported. Add support to more Linux flavours and Windows
+- Handle package dependencies when installing through S3
+- Test edge cases
+
+## Getting Started
+
+### Requirements
+
+- [Terraform](https://developer.hashicorp.com/terraform/tutorials/aws-get-started/install-cli)
+- Guardrails mods:
+  - [@turbot/aws-ssm](https://hub-guardrails-turbot-com-git-development-turbot.vercel.app/aws/mods/aws-ssm)
+- AWS resources to run the stack control:
+  - S3 bucket to store packages
+  - An IAM role for SSM document execution with the following configured permissions:
+
+    ```json
+    {
+      "Version": "2012-10-17",
+      "Statement": [
+        {
+          "Effect": "Allow",
+          "Action": [
+            "ssm:DescribeInstanceInformation",
+            "ssm:ListCommandInvocations",
+            "ssm:ListCommands",
+            "ssm:SendCommand"
+          ],
+          "Resource": "*"
+        },
+        {
+          "Effect": "Allow",
+          "Action": [
+            "s3:PutObject"
+          ],
+          "Resource": "*"
+        }
+      ]
+    }
+    ```
+
+  - An IAM role for the EC2 instance with the following configured permissions:
+
+    ```json
+    {
+      "Version": "2012-10-17",
+      "Statement": [
```

Comment:
Do we need multiple statements or could we just have 1 (since it's only 1 perm in the 2nd)?

Also, we should probably have a `Sid` for the statement.

Reply:
Updated.